include/text.h

```diff
@@ -0,0 +1,4592 @@
+#define TEXT_TEXTID_12C5 0x12C5
+#define TEXT_TEXTID_12C6 0x12C6
+#define TEXT_TEXTID_12C7 0x12C7
+#define TEXT_TEXTID_12C8 0x12C8
+#define TEXT_TEXTID_12C9 0x12C9
+#define TEXT_TEXTID_12CA 0x12CA
+#define TEXT_TEXTID_12CB 0x12CB
+#define TEXT_TEXTID_12CC 0x12CC
+#define TEXT_TEXTID_12CD 0x12CD
+#define TEXT_TEXTID_12CE 0x12CE
+#define TEXT_TEXTID_12CF 0x12CF
+#define TEXT_TEXTID_12D0 0x12D0
+#define TEXT_TEXTID_12D1 0x12D1
+#define TEXT_TEXTID_12D2 0x12D2
+#define TEXT_TEXTID_12D3 0x12D3
+#define TEXT_TEXTID_12D4 0x12D4
+#define TEXT_TEXTID_12D5 0x12D5
+#define TEXT_TEXTID_12D6 0x12D6
+#define TEXT_TEXTID_12D7 0x12D7
+#define TEXT_TEXTID_12D8 0x12D8
+#define TEXT_TEXTID_12D9 0x12D9
+#define TEXT_TEXTID_12DA 0x12DA
+#define TEXT_TEXTID_12DB 0x12DB
+#define TEXT_TEXTID_12DC 0x12DC
+#define TEXT_TEXTID_12DD 0x12DD
+#define TEXT_TEXTID_12DE 0x12DE
+#define TEXT_TEXTID_12DF 0x12DF
+#define TEXT_TEXTID_12E0 0x12E0
+#define TEXT_TEXTID_12E3 0x12E3
+#define TEXT_TEXTID_12E4 0x12E4
+#define TEXT_TEXTID_12E5 0x12E5
+#define TEXT_TEXTID_12E6 0x12E6
+#define TEXT_TEXTID_12E7 0x12E7
+#define TEXT_TEXTID_12E8 0x12E8
+#define TEXT_TEXTID_12E9 0x12E9
+#define TEXT_TEXTID_12EA 0x12EA
+#define TEXT_TEXTID_12EB 0x12EB
+#define TEXT_TEXTID_12EC 0x12EC
+#define TEXT_TEXTID_12ED 0x12ED
+#define TEXT_TEXTID_12EE 0x12EE
+#define TEXT_TEXTID_12EF 0x12EF
+#define TEXT_TEXTID_12F0 0x12F0
+#define TEXT_TEXTID_12F1 0x12F1
+#define TEXT_TEXTID_12F2 0x12F2
+#define TEXT_TEXTID_12F3 0x12F3
+#define TEXT_TEXTID_12F4 0x12F4
+#define TEXT_TEXTID_12F5 0x12F5
+#define TEXT_TEXTID_12F6 0x12F6
+#define TEXT_TEXTID_12F7 0x12F7
+#define TEXT_TEXTID_12F8 0x12F8
+#define TEXT_TEXTID_12F9 0x12F9
+#define TEXT_TEXTID_12FA 0x12FA
+#define TEXT_TEXTID_12FB 0x12FB
+#define TEXT_TEXTID_12FC 0x12FC
+#define TEXT_TEXTID_12FD 0x12FD
+#define TEXT_TEXTID_12FE 0x12FE
+#define TEXT_TEXTID_12FF 0x12FF
+#define TEXT_TEXTID_1300 0x1300
+#define TEXT_TEXTID_1301 0x1301
+#define TEXT_TEXTID_1302 0x1302
+#define TEXT_TEXTID_1303 0x1303
+#define TEXT_TEXTID_1304 0x1304
+#define TEXT_TEXTID_1305 0x1305
+#define TEXT_TEXTID_1306 0x1306
+#define TEXT_TEXTID_1307 0x1307
+#define TEXT_TEXTID_1308 0x1308
+#define TEXT_TEXTID_1309 0x1309
+#define TEXT_TEXTID_130A 0x130A
+#define TEXT_TEXTID_130B 0x130B
+#define TEXT_TEXTID_130C 0x130C
+#define TEXT_TEXTID_130D 0x130D
+#define TEXT_TEXTID_130E 0x130E
+#define TEXT_TEXTID_130F 0x130F
+#define TEXT_TEXTID_1310 0x1310
+#define TEXT_TEXTID_1311 0x1311
+#define TEXT_TEXTID_1312 0x1312
+#define TEXT_TEXTID_1313 0x1313
+#define TEXT_TEXTID_1314 0x1314
+#define TEXT_TEXTID_1315 0x1315
+#define TEXT_TEXTID_1316 0x1316
+#define TEXT_TEXTID_1317 0x1317
+#define TEXT_TEXTID_1318 0x1318
+#define TEXT_TEXTID_1319 0x1319
+#define TEXT_TEXTID_131A 0x131A
+#define TEXT_TEXTID_131B 0x131B
+#define TEXT_TEXTID_131C 0x131C
+#define TEXT_TEXTID_131D 0x131D
+#define TEXT_EN_TALK_GIBUD_LEAVE_IT 0x1388
+#define TEXT_EN_TALK_GIBUD_AWAY 0x1389
+#define TEXT_EN_TALK_GIBUD_NO_REMORSE 0x138A
```

Comment:
```suggestion
#define TEXTID_131B 0x131B
#define TEXTID_131C 0x131C
#define TEXTID_131D 0x131D
#define TEXTID_TALKGIBUD_LEAVE_IT 0x1388
#define TEXTID_TALKGIBUD_AWAY 0x1389
#define TEXTID_TALKGIBUD_NO_REMORSE 0x138A
```
A couple of comments. The code above is all a need though so my thoughts will be summarized here:
- I'm still not entirely sure if every text in the game can be summarized by a few small words. Are defines feasible for every text with the `TEXTID`/`ACTORNAME` prefix be uniquely characterized by only two-three words?
- I'm not sure I fully understand the text system, but with these large scale Ids, usually there's more information than just a simple list in these numbers. Are messages in the `0x0XXX` treated differently than textId's in the `0x1000` vs `0x2000` vs `0x3000`. They certainly are for sfx's, which is captured inside the define name. Maybe this is already known in OoT (since message has been decompiled) and maybe it's not the case. I genuinely don't know. But it's something to ask and to look into.
-  Also, is every text Id used by an actor? Are textId's used by multiple actors? How would the abbreviation happen there? What about unused texts. Is every textId used in the game?
- I think its important to distinguish `TEXT` (which I would interpret as a string) and `TEXTID` (which I would interpret as a lookup number for `TEXT`). So the prefix should be `TEXTID_` and that should be enough (although maybe `TEXT_` would be fine too). But `TEXTID_TEXT_` is redundant
- I think the style, assuming this fits, would be `TEXTID_ACTORNAME_DESCRIPTION_OF_TEXT`. I don't think the `EN` is needed, these are going to get long enough as is. And building on my point above, this assumes each textId lines up nicely with 1 textId matches to 1 actor.
- All the above being said, I don't have a better solution than what's being proposed here. I feel something like this might be worth coordinating with OoT, they've had a lot more time to think about this problem. 

Just my thoughts :)

Reply:
Some thoughts in response to your thoughts:
- No, absolutely not. Just look at my Jg PR, which uses significantly more words to define text IDs. They're really only this short because Talk_Gibud's messages are easy to summarize.
- There are *some* patterns in how texts are arranged. For example, 0x02XX seems to be all the non-enemy Tatl hints, and 0x19XX seems to be all the enemy Tatl hints (It looks like to get the hint text ID for any enemy, take `this->actor.hintId` and add 0x1900 to it). But I'm not sure if the game actually treats the IDs any differently, or if the text data itself is what makes the text different. Just looking at the texts, 0x1FFF and 0x2000 are both used in the Happy Mask Salesman's spiel about getting Majora's Mask back, which suggests at least that the game doesn't really care about 0x1XXX vs 0x2XXX. It would be good to actually know, though.
- To break down this one thought into multiple responses: no, not every text ID ends up actually being used by an actor. Jg has a text that he never says. In his specific case, it's useful to keep a define around, though, because the code checks to see if he ever says it. I'm sure there are text IDs in this game where there is actual text data, but nothing ever references it, though I can't think of an example off the top of my head. There are some Japanese Tatl hints that look completely unused, and never got translated into English, though.
- Yes, there are text IDs used by multiple actors. One is actually in this PR; Talk_Gibud uses the text ID for telling the player how to open the item menu. In this case, it just has no prefix. I suppose it could have a `GLOBAL` prefix for texts like this, but that seems wasteful. In terms of texts that are used across multiple actors that *aren't* just global strings...I'm not sure those exist, but I haven't looked into it too deeply.
- No, not every text ID is used in the game. There are lots of gaps in the list of text; one example is that there is text for 0xFBF, then it jumps to 0x1004. If I did my script right, these unused values should not exist in this file; I can see the FBF -> 1004 jump in my list of defines, but I didn't check every single one of them.
- RE: naming, yeah, I agree with all your points. I can go change what I currently have to match.
- And yeah, it would be nice if OoT and MM decided on the same system. They also have message decompiled, so a lot of the unknowns I talk about above aren't unknown there.

I think looking at my Jg PR helps to explain my motivation behind doing this. The current tutorial on documentation advocates using comments next to text IDs to document how an actor works; for Jg, I don't think this is sustainable at all, because many text IDs are used multiple times and/or in giant conditional statements. Actors that use a lot of text IDs (like Jg) are almost indecipherable unless the text ID literals are swapped out with *something* that has any sort of meaning.